listServices method throwing exception

## Description

Whilst attempting to use the listServices method, an exception is produced

## How to reproduce

```
DockerClient docker = DefaultDockerClient.fromEnv().build();
List<Service> services = docker.listServices();
docker.close();
```

## What do you expect

I expected it to work without an exception.

## What happened instead

Exception thrown.

## Software:

- `docker version`: 1.13.1
- Spotify's docker-client version: 8.1.2

## Full backtrace

```
Exception in thread "main" com.spotify.docker.client.exceptions.DockerRequestException: Request error: GET https://blah.com:12345/services: 200
	at com.spotify.docker.client.DefaultDockerClient.propagate(DefaultDockerClient.java:2165)
	at com.spotify.docker.client.DefaultDockerClient.request(DefaultDockerClient.java:2094)
	at com.spotify.docker.client.DefaultDockerClient.listServices(DefaultDockerClient.java:1680)
	at edu.bristol.webfarmrepdockerclientexperiment.Test.main(Test.java:21)
Caused by: javax.ws.rs.client.ResponseProcessingException: com.fasterxml.jackson.databind.JsonMappingException: Can not construct instance of com.spotify.docker.client.messages.mount.Driver, problem: abstract types either need to be mapped to concrete types, have custom deserializer, or be instantiated with additional type information
 at [Source: org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream@582571ea; line: 1, column: 4164] (through reference chain: java.util.ArrayList[2]->com.spotify.docker.client.messages.swarm.Service["Spec"]->com.spotify.docker.client.messages.swarm.ServiceSpec["TaskTemplate"]->com.spotify.docker.client.messages.swarm.TaskSpec["ContainerSpec"]->com.spotify.docker.client.messages.swarm.ContainerSpec["Mounts"]->java.util.ArrayList[1]->com.spotify.docker.client.messages.mount.Mount["VolumeOptions"]->com.spotify.docker.client.messages.mount.VolumeOptions["DriverConfig"])
	at org.glassfish.jersey.client.JerseyInvocation.translate(JerseyInvocation.java:869)
	at org.glassfish.jersey.client.JerseyInvocation.access$800(JerseyInvocation.java:92)
	at org.glassfish.jersey.client.JerseyInvocation$6.completed(JerseyInvocation.java:836)
	at org.glassfish.jersey.client.ClientRuntime.processResponse(ClientRuntime.java:198)
	at org.glassfish.jersey.client.ClientRuntime.access$300(ClientRuntime.java:79)
	at org.glassfish.jersey.client.ClientRuntime$2.run(ClientRuntime.java:180)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:340)
	at org.glassfish.jersey.client.ClientRuntime$3.run(ClientRuntime.java:210)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Can not construct instance of com.spotify.docker.client.messages.mount.Driver, problem: abstract types either need to be mapped to concrete types, have custom deserializer, or be instantiated with additional type information
 at [Source: org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream@582571ea; line: 1, column: 4164] (through reference chain: java.util.ArrayList[2]->com.spotify.docker.client.messages.swarm.Service["Spec"]->com.spotify.docker.client.messages.swarm.ServiceSpec["TaskTemplate"]->com.spotify.docker.client.messages.swarm.TaskSpec["ContainerSpec"]->com.spotify.docker.client.messages.swarm.ContainerSpec["Mounts"]->java.util.ArrayList[1]->com.spotify.docker.client.messages.mount.Mount["VolumeOptions"]->com.spotify.docker.client.messages.mount.VolumeOptions["DriverConfig"])
```

--- a/src/main/java/com/spotify/docker/client/messages/mount/Driver.java
+++ b/src/main/java/com/spotify/docker/client/messages/mount/Driver.java
@@ -24,19 +24,24 @@ import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.ANY;
 import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.NONE;
 
 import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.auto.value.AutoValue;
 
 import com.google.common.collect.ImmutableMap;
 import java.util.Map;
 
+import javax.annotation.Nullable;
+
 @AutoValue
 @JsonAutoDetect(fieldVisibility = ANY, getterVisibility = NONE, setterVisibility = NONE)
 public abstract class Driver {
 
+  @Nullable
   @JsonProperty("Name")
   public abstract String name();
 
+  @Nullable
   @JsonProperty("Options")
   public abstract ImmutableMap<String, String> options();
 
@@ -59,5 +64,15 @@ public abstract class Driver {
 
   public static Driver.Builder builder() {
     return new AutoValue_Driver.Builder();
+  }
+
+  @JsonCreator
+  static Driver create(
+      @JsonProperty("Name") final String name,
+      @JsonProperty("Options") final Map<String, String> options) {
+    return builder()
+        .name(name)
+        .options(options)
+        .build();
   }
 }

--- a/src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java
+++ b/src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java
@@ -4442,7 +4442,9 @@ public class DefaultDockerClientTest {
             .image("alpine")
             .command(new String[] {"ping", "-c1000", "localhost"})
             .mounts(Mount.builder()
-                .volumeOptions(VolumeOptions.builder().build())
+                .volumeOptions(VolumeOptions.builder()
+                    .driverConfig(com.spotify.docker.client.messages.mount.Driver.builder().build())
+                    .build())
                 .bindOptions(BindOptions.builder().build())
                 .tmpfsOptions(TmpfsOptions.builder().build())
                 .build())
@@ -4496,8 +4498,18 @@ public class DefaultDockerClientTest {
 
     final ContainerSpec containerSpec = actualServiceSpec.taskTemplate().containerSpec();
     assertThat(containerSpec.labels(), equalTo(Collections.<String, String>emptyMap()));
+
     assertThat(containerSpec.mounts().size(), equalTo(1));
-    assertThat(containerSpec.mounts().get(0).type(), equalTo("bind"));
+    final Mount mount = containerSpec.mounts().get(0);
+    assertThat(mount.type(), equalTo("bind"));
+
+    final VolumeOptions volumeOptions = mount.volumeOptions();
+    assertThat(volumeOptions.noCopy(), nullValue());
+    assertThat(volumeOptions.labels(), nullValue());
+
+    final com.spotify.docker.client.messages.mount.Driver driver = volumeOptions.driverConfig();
+    assertThat(driver.name(), nullValue());
+    assertThat(driver.options(), nullValue());
 
     final RestartPolicy restartPolicy = actualServiceSpec.taskTemplate().restartPolicy();
     assertThat(restartPolicy.condition(), equalTo(RESTART_POLICY_ANY));

